avionics/common/include/state_events.h

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include "HAL/time.h"
+#include "radio.h"
+#include "state_interface.h"
+
+namespace State {
+    const std::unordered_map<StateId, EventId> STATE_CHANGE_EVENTS = {
+        {StateId::ARMED, EventId::ARMED},
+        {StateId::STANDBY, EventId::DISARMED},
+        {StateId::POWERED_ASCENT, EventId::LAUNCH},
+        {StateId::MACH_LOCK, EventId::MACH_LOCK_ENTER},
+        {StateId::ASCENT_TO_APOGEE, EventId::MACH_LOCK_EXIT},
```

Comment:
Was just playing around with this branch while working on the full flight sim stuff and I noticed this might need to review this with @jesseli2002 

The flight goes: standby -> ascent to apogee -> mach lock -> ascent to apogee

Looks like the current map assumes `ASCENT_TO_APOGEE` occurs only after `MACH_LOCK`.

Also because of this non-linear state progression, we might not be able to use a map and might need to resort to something else (such as sending the events from the actual states).

Reply:
yeah that's part of why I wanted to overhaul state machine design - throwing out state change events should be done by the state machine

Reply:
that wouldn't be the only non-linear bit - there's also arm/disarm that switches back and forth, and both standby & armed can go to ascent

although I don't see why you wouldn't want to keep the event mapping - just indicate events on state entry in all cases

Reply:
Yeah good point, that seems like the easiest/simplest way to do it for now at least.